src/lib/pipeline/m2/material/index.js

```diff
@@ -257,6 +270,58 @@ class M2Material extends THREE.ShaderMaterial {
     }
   }
 
+  registerAnimations(def) {
+    const { uvAnimations, transparencyAnimations, vertexColorAnimation } = def;
+
+    this.registerUVAnimations(uvAnimations);
+    this.registerTransparencyAnimations(transparencyAnimations);
+    this.registerVertexColorAnimation(vertexColorAnimation);
+  }
+
+  registerUVAnimations(uvAnimationIndices) {
+    const { animations, uvAnimationValues } = this.m2;
+
+    uvAnimationIndices.forEach((uvAnimationIndex, opIndex) => {
+      const target = this.uniforms.animatedUVs;
+      const source = uvAnimationValues[uvAnimationIndex];
+
+      animations.on('update', () => {
+        target.value[opIndex] = source.matrix;
+      });
```

Comment:
Not quite sure I follow what you mean by "listen for update for every single index" -- a material only listens to update for the indices that are relevant to the material (defined by uvAnimationIndices).

You're probably right that this is going to prevent GC. Do listeners have handles at all? Could I deregister if I kept track of the handle?